### ISSUE TITLE
ActiveRecord support

### ISSUE THREAD

**Issue by pleerock** [typeorm/typeorm#480]
Its really cool if typeorm can support both data mapper and active record patterns.

Example:

```ts
@Entity()
export class Post extends EntityModel {

    @PrimaryGeneratedColumn()
    id: number;

    @Column()
    title: string;

    @Column()
    text: string;

}
```

And then:

```ts
const post = new Post();
post.title = "About ActiveRecord";
post.text = "Huge discussion how good or bad ActiveRecord is.";
await post.save();

const loadedPost = await Post.findOneById(1);
```

**Comment by pleerock** [comment]
AR supported was added in ce7f36b162c1282082fab37afa7a3694289f7890 however there is an [issue with TypeScript](https://github.com/Microsoft/TypeScript/issues/15946) I have because of which `Post.findOneById(1)` returns `any` instead of `Post`

**Comment by pleerock** [comment]
implemented and is available in `0.1.0-alpha.2`. Need to add docs later on.
